lib/fs.js

```diff
@@ -1112,14 +1114,14 @@ function writeAll(fd, buffer, offset, length, position, callback) {
 }
 
 fs.writeFile = function(path, data, options, callback) {
-  var callback = maybeCallback(arguments[arguments.length - 1]);
+  callback = maybeCallback(arguments[arguments.length - 1]);
```

Comment:
@ChALkeR I tried something like this

    function containsWith(a_) {
        var a = arguments[0] + 1;
    }

    function containsWith(a) {
        a = arguments[0] + 1;
    }

but it still says the function is not optimized for both the cases.